src/schema/sources.ts

```diff
@@ -992,24 +994,27 @@ export const isPrivilegedMember = async (
   return sourceRoleRank[sourceMember.role] >= sourceRoleRank.moderator;
 };
 
+type PostPermissions = SourcePermissions.Post | SourcePermissions.PostRequest;
+
 export const canPostToSquad = (
   ctx: Context,
   squad: SquadSource,
   sourceMember: SourceMember | null,
+  permission: PostPermissions = SourcePermissions.Post,
 ): boolean => {
   if (!sourceMember) {
     return false;
   }
 
   const memberRank = sourceRoleRank[sourceMember.role];
 
-  if (
-    memberRank >= sourceRoleRank.moderator &&
-    memberRank >= squad.memberPostingRank
-  )
-    return true;
+  if (squad.moderationRequired) {
+    if (memberRank === sourceRoleRank.member) {
+      return permission === SourcePermissions.PostRequest;
+    }
+  }
```

Comment:
Is this correct, why `===`?

Reply:
asking because condition looks completely different then what it was

Reply:
My idea with this condition was to check if user is a moderator or higher (admin), and let them bypass the moderation